apps/authentication-portal/src/main/webapp/login.jsp

```diff
@@ -408,17 +415,43 @@
                                             </form>
 
                                             <script>
-                                                window.onload = function callGoogleOneTap() {
-                                                    google.accounts.id.initialize({
-                                                        client_id: "<%=Encode.forJavaScriptAttribute(GOOGLE_CLIENT_ID)%>",
-                                                        prompt_parent_id: "google_parent",
-                                                        cancel_on_tap_outside: false,
-                                                        nonce: "<%=Encode.forJavaScriptAttribute(request.getParameter("sessionDataKey"))%>",
-                                                        callback: handleCredentialResponse
-                                                    });
-                                                    google.accounts.id.prompt((notification) => {
-                                                        onMoment(notification);
-                                                    });
+                                                if (navigator) {
+                                                    var userAgent = navigator.userAgent;
+                                                    var browserName = void 0;
+                                                    var restrictedBrowsersForGOT = "<%=restrictedBrowsersForGOT%>";
+
+                                                    if (userAgent.match(/chrome|chromium|crios/i)) {
+                                                        browserName = "chrome";
+                                                    } else if (userAgent.match(/firefox|fxios/i)) {
+                                                        browserName = "firefox";
+                                                    } else if (userAgent.match(/safari/i)) {
+                                                        browserName = "safari";
+                                                    } else if (userAgent.match(/opr\//i)) {
+                                                        browserName = "opera";
+                                                    } else if (userAgent.match(/edg/i)) {
+                                                        browserName = "edge";
+                                                    } else {
+                                                        browserName = "No browser detection";
+                                                    }
+
+                                                    if (restrictedBrowsersForGOT !== null
+                                                        && restrictedBrowsersForGOT !== ''
```

Comment:
Do we need these checks? `restrictedBrowsersForGOT` can't be null according to the definition right?

Reply:
Currently no. However it depends on EndpointConfigManager which is a separate module. The code patterns can be modified based on the module. I believe we should not depend on a separate module for this level verifications. WDYT ?